components/backdrop/backdrop.js

```diff
@@ -0,0 +1,151 @@
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { getComposedChildren, getComposedParent } from '../../helpers/dom.js';
+
+let count = 0;
+let overflow = null;
+
+class Backdrop extends LitElement {
+
+	static get properties() {
+		return {
+			show: { type: Boolean, reflect: true },
+			forTarget: { type: String, attribute: 'for-target' },
+			_state: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return [ css`
+			:host {
+				background-color: var(--d2l-color-regolith);
+				height: 0;
+				left: 0;
+				opacity: 0;
+				position: fixed;
+				top: 0;
+				transition: opacity 200ms ease-in;
+				width: 0;
+				z-index: 999;
+			}
+			:host([_state="showing"]), :host([_state="hiding"]) {
+				height: 100%;
+				width: 100%;
+			}
+			:host([_state="showing"]) {
+				opacity: 0.7;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.show = false;
+		this._state = null;
+	}
+
+	attributeChangedCallback(name, oldval, newval) {
+		super.attributeChangedCallback(name, oldval, newval);
+
+		if (name === 'show' && oldval !== newval) {
+			if (this.show) {
+
+				if (count === 0) {
+					overflow = document.body.style.overflow;
+					document.body.style.overflow = 'hidden';
+				}
+				count++;
+
+				this._hiddenElements = hideAccessible(this.parentNode.querySelector(`#${this.forTarget}`));
+				this._state = 'showing';
+			} else {
+				const transitionEnd = () => {
+					this.removeEventListener('transitionend', transitionEnd);
+
+					count--;
+					if (count === 0) {
+						document.body.style.overflow = overflow;
+						overflow = null;
+					}
+
+					showAccessible(this._hiddenElements);
```

Comment:
Wondering... should we revert things back to the way they were in the `disconnectedCallback`? Just thinking of the case where the dialog is removed from the DOM, which should remove its backdrop, which should undo the things it did.

Reply:
Yeah probably.  I'm going to see what I can do with this.  In theory it could be reconnected too.